inst/python_cmd.m

```diff
@@ -138,11 +138,20 @@
   end
 
   if (strcmp(A{1}, 'COMMAND_ERROR_PYTHON'))
-    errlineno = A{3} - LinesBeforeCmdBlock - 1;
-    errstr = sprintf(['Python raised exception, maybe at line %d of code ' ...
-                      'block (assuming popen2)\n' ...
-                      '%s'], errlineno, A{2});
-    error('OctSymPy:python_cmd:exception', errstr);
+
+    ex_popen2 = exist('popen2') > 1;
+    ipc_s = sympref('ipc'); 
+
+    if strcmp(ipc_s, 'popen2') || strcmp(ipc_s, 'sysoneline') || (strcmp(ipc_s, 'default') && ex_popen2)
+      errlineno = A{3} - line_error_python('get');
+    elseif strcmp(ipc_s, 'system') || strcmp(ipc_s, 'systmpfile') || (strcmp(ipc_s, 'default') && !ex_popen2)
+      errlineno = A{3} - line_error_python('get') - (nargin - 1)*2;
```

Comment:
this assumes every input takes exactly two lines?  I'm not sure that's true: maybe list or struct/dict breaks this?  Did you test with struct?

Even if true now, might not be true in future.